carbon-design-system/carbon
# feat: implement codemod ibm-products-update-userprofileimage

Closes [6308](https://github.com/carbon-design-system/ibm-products/issues/6308)

This PR updates the @carbon/upgrade package with a new transform ibm-products-update-userprofileimage, to update the  UserProfileImage to UserAvatar.

#### Changelog

**New**

- New transform packages/upgrade/transforms/ibm-products-update-userprofileimage.js
- Include fixtures in the packages/upgrade/transforms/__testfixtures__/ibm-products-update-userprofileimage.input.js and packages/upgrade/transforms/__testfixtures__/ibm-products-update-userprofileimage.output.js files
- Include tests packages/upgrade/transforms/__tests__/ibm-products-update-userprofileimage-test.js
- Implement a sample packages/upgrade/fixtures/ibm-products-update-userprofileimage-sample.jsx to test the Codemod locally

**Changed**

- Update packages/upgrade/src/upgrades.js with a new migration entry ibm-products-update-userprofileimage

#### Testing / Reviewing

Try out the Codemod locally by:
- Pulling down this PR
- Run cd packages/upgrade
- Run yarn build to build the CLI
- Run `./bin/carbon-upgrade.js migrate list` to verify a new entry ibm-products-update-userprofileimage (Rewrites UserProfileImage to UserAvatar) is present
- Run `npx jest transforms/__tests__/ibm-products-update-userprofileimage-test.js` and verify test result
- Run `./bin/carbon-upgrade.js migrate ibm-products-update-userprofileimage fixtures/ibm-products-update-userprofileimage-sample.jsx -w` to try the Codemod. ( Select the /carbon/packages/upgrade as workspace)
